WCFDataService/Client/System/Data/Services/Client/DataServiceContext.cs

```diff
@@ -77,6 +77,34 @@ public enum UndeclaredPropertyBehavior
     }
 
     /// <summary>
+    /// Indicates how DataServiceContext will check for uninitialized properties on the proxy types.
+    /// Proxies with lazy initialized properties (initialized in the property getter) can specify the naming convention for the backing fields so the DataServiceContext does not trigger initialization when not desired.
+    /// If no backing fields can be found, the property getters are used falling back to the "old" behaviour.
+    /// </summary>
+    public enum ProxyBackingFieldNamingConvention
+    {
+        /// <summary>
+        /// The default value, the DataServiceContext will not try to find backing fields but use the property getters. This is the "old" behaviour.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// The DataServiceContext will first try ProxyBackingFieldNamingConvention.Underscores then ProxyBackingFieldNamingConvention.CamelCasing.
+        /// </summary>
+        Auto = 1,
```

Comment:
Yes you are right. The only scenario here would be if the (handwritten) proxy code were inconsistent in their naming conventions and used underscores for some fields and camel casing for others. I think it would be ok to require constancy from the proxy code. I'll remove the Auto option.